pkg/libs/endpointaccessible/endpoint_accessible_controller.go

```diff
@@ -155,22 +174,30 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 	return utilerrors.NewAggregate(errors)
 }
 
-func (c *endpointAccessibleController) buildTLSClient() (*http.Client, error) {
+func (c *endpointAccessibleController) buildTLSClient() (*http.Client, bool, error) {
 	transport := &http.Transport{
 		Proxy: http.ProxyFromEnvironment,
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: true,
 		},
 	}
+
+	tlsChanged := false
 	if c.getTLSConfigFn != nil {
 		tlsConfig, err := c.getTLSConfigFn()
 		if err != nil {
-			return nil, err
+			return nil, false, err
 		}
 		transport.TLSClientConfig = tlsConfig
+
+		// these are the fields that are set by our getTLSConfigFn funcs
+		tlsChanged = c.lastServerName != tlsConfig.ServerName || !tlsConfig.RootCAs.Equal(c.lastCA)
```

Comment:
`tlsConfig` is a generic construct, is comparison of just these two attributes good enough?